From: Sascha Hauer <s.hauer@pengutronix.de>
Date: Wed, 20 Jul 2016 13:58:20 +0200
Subject: [PATCH] bootentries: Move struct bootentries to include/boot.h

The boot function prototypes are declared in include/blspec.h. Move them
to include/boot.h where they belong.

Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
---
 include/blspec.h | 21 +--------------------
 include/boot.h   | 20 ++++++++++++++++++++
 2 files changed, 21 insertions(+), 20 deletions(-)

diff --git a/include/blspec.h b/include/blspec.h
index 8a79df5738ad..7d911f030bfa 100644
--- a/include/blspec.h
+++ b/include/blspec.h
@@ -2,21 +2,7 @@
 #define __LOADER_H__
 
 #include <linux/list.h>
-#include <menu.h>
-
-struct bootentries {
-	struct list_head entries;
-	struct menu *menu;
-};
-
-struct bootentry {
-	struct list_head list;
-	struct menu_entry me;
-	char *title;
-	char *description;
-	int (*boot)(struct bootentry *entry, int verbose, int dryrun);
-	void (*release)(struct bootentry *entry);
-};
+#include <boot.h>
 
 struct blspec_entry {
 	struct bootentry entry;
@@ -37,9 +23,4 @@ int blspec_scan_device(struct bootentries *bootentries, struct device_d *dev);
 int blspec_scan_devicename(struct bootentries *bootentries, const char *devname);
 int blspec_scan_directory(struct bootentries *bootentries, const char *root);
 
-int bootentries_add_entry(struct bootentries *entries, struct bootentry *entry);
-
-#define bootentries_for_each_entry(bootentries, entry) \
-	list_for_each_entry(entry, &bootentries->entries, list)
-
 #endif /* __LOADER_H__ */
diff --git a/include/boot.h b/include/boot.h
index da40ac2ece29..e0a61c564409 100644
--- a/include/boot.h
+++ b/include/boot.h
@@ -2,6 +2,7 @@
 #define __BOOT_H
 
 #include <of.h>
+#include <menu.h>
 #include <environment.h>
 
 #ifdef CONFIG_FLEXIBLE_BOOTARGS
@@ -19,4 +20,23 @@ static inline int linux_bootargs_overwrite(const char *bootargs)
 }
 #endif
 
+struct bootentries {
+	struct list_head entries;
+	struct menu *menu;
+};
+
+struct bootentry {
+	struct list_head list;
+	struct menu_entry me;
+	char *title;
+	char *description;
+	int (*boot)(struct bootentry *entry, int verbose, int dryrun);
+	void (*release)(struct bootentry *entry);
+};
+
+int bootentries_add_entry(struct bootentries *entries, struct bootentry *entry);
+
+#define bootentries_for_each_entry(bootentries, entry) \
+	list_for_each_entry(entry, &bootentries->entries, list)
+
 #endif /* __BOOT_H */
